src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinHttpWebSocket.cs

```diff
@@ -441,6 +448,12 @@ private Task<bool> InternalReceiveAsync(ArraySegment<byte> buffer)
                 Abort();
             }
 
+            // If the task has completed, there's no need to force continuations to run async
+            if (_operation.TcsReceive.Task.Status == TaskStatus.RanToCompletion)
+            {
+                return Task.FromResult(_operation.TcsReceive.Task.GetAwaiter().GetResult());
+            }
```

Comment:
I'm still not understanding.  If _operation.TcsReceive.Task.Status == TaskStatus.RanToCompletion, then calling Wait on _operation.TcsReceive.Task (which is what's returned below) is a nop.  So why return Task.CompletedTask rather than returning _operation.TcsReceive.Task?  The only difference I can see is that, with your change, _operation.TcsReceive.Task was created with RunContinuationsAsynchronously, whereas Task.CompletedTask wasn't.  But since both tasks would already be completed in this case by the time the method returns, await'ing either of them wouldn't register a continuation, and so there wouldn't be any difference.  The only difference would be if you registered a synchronous continuation with the returned already-completed task, hence my question.